tests/foreman/cli/test_auth.py

```diff
@@ -0,0 +1,417 @@
+        """Check if session stays up after failed login attempt
+
+        :id: 6c4d5c4c-eff0-411b-829f-0c2f2ec26132
+
+        :BZ: 1465552
+
+        :Steps:
+
+            1. Set use_sessions
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Run login with invalid credentials
+
+        :expectedresults: The session is unchanged
+
+        """
+        result = configure_sessions()
+        self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+        Auth.login({'username': self.uname_admin, 'password': self.password})
+        result = Auth.status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        with self.assertRaises(CLIReturnCodeError):
+            Org.list(pass_credentials=False)
+        try:
+            Auth.login({
+                'username': self.uname_viewer,
+                'password': gen_string('alpha')})
+        except:
+            result = Auth.status()
+            self.assertIn(
+                LOGEDIN_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+            with self.assertRaises(CLIReturnCodeError):
+                Org.list(pass_credentials=False)
+
+    @skip_if_bug_open('bugzilla', '1471099')
+    @tier1
+    def test_positive_session_preceeds_saved_credentials(self):
+        """Check if enabled session is mutually exclusive with
+        saved credentials in hammer config
+
+        :id: e4277298-1c24-494b-84a6-22f45f96e144
+
+        :BZ: 1471099
+
+        :Steps:
+
+            1. Set use_sessions, set usernam and password,
+               set short expiration time
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Wait until session expires
+
+        :expectedresults: Session expires after specified time
+            and saved credentials are not applied
+
+        """
+        try:
+            idle = Settings.list({'search': 'name=idle_timeout'})[0][u'value']
+            Settings.set({'name': 'idle_timeout', 'value': 1})
+            result = configure_sessions(add_default_creds=True)
+            self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+            Auth.login({
+                'username': self.uname_admin,
+                'password': self.password
+            })
+            result = Auth.status()
+            self.assertIn(
+                LOGEDIN_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+            # list organizations without supplying credentials
+            with self.assertNotRaises(CLIReturnCodeError):
+                Org.list(pass_credentials=False)
+            # wait until session expires
+            sleep(70)
+            with self.assertRaises(CLIReturnCodeError):
+                Org.list(pass_credentials=False)
```

Comment:
Hammer will ask for new credentials interactively, please make sure the code can cope with that.